tools/android/packaging/Makefile.in

```diff
@@ -153,17 +153,15 @@ libs: $(PREFIX)/lib/@APP_NAME_LC@/lib@APP_NAME_LC@.so
 	echo "directory $(TOOLCHAIN)/sysroot/usr/include $(NDKROOT)/sources/android/native_app_glue" \
              "$(NDKROOT)/sources/cxx-stl/gnu-libstdc++/$(GCC_VERSION)/include $(CORE_SOURCE_DIR)  $(PREFIX)/include jni" >> ./xbmc/lib/$(CPU)/gdb.setup
 
-xbmc/classes.dex: res
+java: res
 	mkdir -p xbmc/java/$(APP_PACKAGE_DIR) xbmc/java/$(APP_PACKAGE_DIR)/interfaces xbmc/obj
 	@cp xbmc/src/*.java xbmc/java/$(APP_PACKAGE_DIR)/
 	@cp xbmc/src/interfaces/*.java xbmc/java/$(APP_PACKAGE_DIR)/interfaces/
-	@javac -classpath $(SDKROOT)/platforms/$(SDK_PLATFORM)/android.jar:xbmc/obj -d xbmc/obj -sourcepath xbmc/src xbmc/gen/$(APP_PACKAGE_DIR)/*.java xbmc/java/$(APP_PACKAGE_DIR)/*.java xbmc/java/$(APP_PACKAGE_DIR)/interfaces/*.java $(JAVAC_EXTRA_ARGS)
-	@$(DX) --dex --output=xbmc/classes.dex xbmc/obj xbmc/lib
 
-package: libs python xbmc/classes.dex
-	@echo "Creating package..."
-	@cp images/@APP_NAME_LC@app-debug-skeleton.apk images/@APP_NAME_LC@app-debug-$(CPU)-unaligned.apk
-	@cd xbmc; $(ZIP) -r -q ../images/@APP_NAME_LC@app-debug-$(CPU)-unaligned.apk lib/$(CPU) assets classes.dex
+package: libs python java
+	@echo "Gradle build..."
+	ANDROID_HOME=$(SDKROOT) ./gradlew assemble$(BUILD_TYPE)Unsigned
+	@cp xbmc/build/outputs/apk/xbmc-$(BUILD_TYPE_LC)Unsigned-unsigned.apk images/@APP_NAME_LC@app-debug-$(CPU)-unaligned.apk
```

Comment:
do we need to keep xbmc-$(BUILD_TYPE_LC)Unsigned-unsigned.apk or can we just mv to new location?

Reply:
Good question. I actually have very little clue about the gradle inners.
We did a cp before (https://github.com/xbmc/xbmc/pull/13263/files#diff-b3e127cc38edd4b1068565100a34c0e0L165) so I just kept the same logic